backend/src/main/java/com/redhat/cloud/notifications/db/session/CommonStateSession.java

```diff
@@ -0,0 +1,99 @@
+package com.redhat.cloud.notifications.db.session;
+
+import io.smallrye.mutiny.Uni;
+import org.hibernate.reactive.mutiny.Mutiny;
+import org.jboss.logging.Logger;
+
+import javax.persistence.PrePersist;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.List;
+
+public interface CommonStateSession {
+
+    public static CommonStateSession withSession(Mutiny.Session session) {
+        return new SessionAdapter(session);
+    }
+
+    public static CommonStateSession withStatelessSession(Mutiny.StatelessSession statelessSession) {
+        return new StatelessSessionAdapter(statelessSession);
+    }
```

Comment:
I'm not sure we need these methods. `SessionAdapter` and `StatelessSessionAdapter` could be moved to the factory.